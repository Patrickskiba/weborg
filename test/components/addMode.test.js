import React from 'react'
import {
  render,
  cleanup,
  fireEvent,
  waitForElement,
  act,
} from 'react-testing-library'
import userEvent from 'user-event'
import 'jest-dom/extend-expect'
import jsdom from 'jsdom'

jest.mock('idb-keyval', () => ({
  keys: () => Promise.resolve([]),
  set: () => {},
}))

jest.mock('dropbox')

jest.mock('../../src/utils/dropbox-files')

describe('add mode tests', () => {
  afterEach(cleanup)

  it('adds a new note when filling out all the fields and clicking the save button', async () => {
    const { StoreProvider: Provider } = require('../../src/components/Store')
    const App = require('../../src/components/App').default
    const { getByLabelText, getByTitle, getByText, debug, container } = render(
      <Provider>
        <App />
      </Provider>
    )

    const addBtn = getByTitle('Add')

    fireEvent.click(addBtn, { button: 1 })

    await waitForElement(() => getByLabelText('Level'))

    const level = getByLabelText('Level')
    const headline = getByLabelText('Headline')
    const content = getByLabelText('Content')

    expect(level.value).toEqual('1')

    expect(headline.value).toEqual('')

    expect(content.value).toEqual('')

    const leveltype = getByLabelText('Level')
    userEvent.type(leveltype, '2')

    const headlinetype = getByLabelText('Headline')
    userEvent.type(headlinetype, 'this is an add for test')

    const contenttype = getByLabelText('Content')
    userEvent.type(contenttype, 'here is some test content for the new note')

    const save = getByTitle('save')

    fireEvent.click(save, { button: 1 })

    await waitForElement(() =>
      getByText('here is some test content for the new note')
    )

    expect(getByText('this is an add for test')).toBeDefined()

    expect(
      getByText('here is some test content for the new note')
    ).toBeDefined()

    expect(container).toMatchSnapshot()
  })
})
