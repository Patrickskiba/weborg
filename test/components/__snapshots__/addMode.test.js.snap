// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add mode tests adds a new note when filling out all the fields and clicking the save button 1`] = `
<div>
  <div
    class="app-container"
  >
    <div>
      <div
        class="headline-row undefined"
        data-testid="headline"
        level="1"
      >
        <div
          class="headline-row-item"
        >
          <div>
            <div
              class="headline-star"
            >
              <svg
                height="16px"
                width="16px"
              >
                <circle
                  cx="50%"
                  cy="50%"
                  r="4px"
                />
              </svg>
            </div>
          </div>
          <div
            class="headline-content"
          >
            <div
              class="no-user-select"
            >
              <span>
                <span>
                  This is a headline note
                </span>
                 
              </span>
            </div>
            <div>
              <div
                class="headline-row undefined"
                data-testid="headline"
                level="2"
              >
                <div
                  class="headline-row-item"
                >
                  <div>
                    <div
                      class="headline-star"
                    >
                      <svg
                        height="16px"
                        width="16px"
                      >
                        <circle
                          cx="50%"
                          cy="50%"
                          r="4px"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    class="headline-content"
                  >
                    <div
                      class="no-user-select"
                    >
                      <span>
                        <span>
                          Headline notes have bulletpoints at the start of a line
                        </span>
                         
                      </span>
                    </div>
                    <div />
                  </div>
                </div>
              </div>
              <div
                class="headline-row undefined"
                data-testid="headline"
                level="2"
              >
                <div
                  class="headline-row-item"
                >
                  <div>
                    <div
                      class="headline-star"
                    >
                      <svg
                        height="16px"
                        width="16px"
                      >
                        <circle
                          cx="50%"
                          cy="50%"
                          r="4px"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    class="headline-content"
                  >
                    <div
                      class="no-user-select"
                    >
                      <span>
                        <span>
                          Add a new headline by clicking the blue plus button
                        </span>
                         
                      </span>
                    </div>
                    <div />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="headline-dashplus"
          >
            <svg
              height="16"
              title="dash-collapse"
              width="16"
            >
              <line
                stroke="black"
                stroke-width="1.5"
                x1="8"
                x2="16"
                y1="8"
                y2="8"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="headline-row undefined"
        data-testid="headline"
        level="1"
      >
        <div
          class="headline-row-item"
        >
          <div>
            <div
              class="headline-star"
            >
              <svg
                height="16px"
                width="16px"
              >
                <circle
                  cx="50%"
                  cy="50%"
                  r="4px"
                />
              </svg>
            </div>
          </div>
          <div
            class="headline-content"
          >
            <div
              class="no-user-select"
            >
              <span>
                <span>
                  Click on a headline to edit it
                </span>
                 
              </span>
            </div>
            <div />
          </div>
        </div>
      </div>
      <div
        class="headline-row undefined"
        data-testid="headline"
        level="1"
      >
        <div
          class="headline-row-item"
        >
          <div>
            <div
              class="headline-star"
            >
              <svg
                height="16px"
                width="16px"
              >
                <circle
                  cx="50%"
                  cy="50%"
                  r="4px"
                />
              </svg>
            </div>
          </div>
          <div
            class="headline-content"
          >
            <div
              class="no-user-select"
            >
              <span>
                <span>
                  To delete a note go to the edit screen and click the options icon in the upper right corner
                </span>
                 
              </span>
            </div>
            <div />
          </div>
        </div>
      </div>
      <div
        class="headline-row undefined"
        data-testid="headline"
        level="1"
      >
        <div
          class="headline-row-item"
        >
          <div>
            <div
              class="headline-star"
            >
              <svg
                height="16px"
                width="16px"
              >
                <circle
                  cx="50%"
                  cy="50%"
                  r="4px"
                />
              </svg>
            </div>
          </div>
          <div
            class="headline-content"
          >
            <div
              class="no-user-select"
            >
              <span>
                <span>
                  A headline's level determines its parent and children
                </span>
                 
              </span>
            </div>
            <div>
              <div
                class="headline-row undefined"
                data-testid="headline"
                level="3"
              >
                <div
                  class="headline-row-item"
                >
                  <div>
                    <div
                      class="headline-star"
                    >
                      <svg
                        height="16px"
                        width="16px"
                      >
                        <circle
                          cx="50%"
                          cy="50%"
                          r="4px"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    class="headline-content"
                  >
                    <div
                      class="no-user-select"
                    >
                      <span>
                        <span>
                          This is a child of the headline above it
                        </span>
                         
                      </span>
                    </div>
                    <div />
                  </div>
                </div>
              </div>
              <div
                class="headline-row undefined"
                data-testid="headline"
                level="3"
              >
                <div
                  class="headline-row-item"
                >
                  <div>
                    <div
                      class="headline-star"
                    >
                      <svg
                        height="16px"
                        width="16px"
                      >
                        <circle
                          cx="50%"
                          cy="50%"
                          r="4px"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    class="headline-content"
                  >
                    <div
                      class="no-user-select"
                    >
                      <span>
                        <span>
                          Click on the dash on the right to collapse a headline's children
                        </span>
                         
                      </span>
                    </div>
                    <div />
                  </div>
                </div>
              </div>
              <div
                class="headline-row undefined"
                data-testid="headline"
                level="2"
              >
                <div
                  class="headline-row-item"
                >
                  <div>
                    <div
                      class="headline-star"
                    >
                      <svg
                        height="16px"
                        width="16px"
                      >
                        <circle
                          cx="50%"
                          cy="50%"
                          r="4px"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    class="headline-content"
                  >
                    <div
                      class="no-user-select"
                    >
                      <span>
                        <span>
                          Notes don't have to be headlines
                        </span>
                         
                      </span>
                    </div>
                    <div>
                      <div
                        class="no-user-select"
                      >
                        <div
                          class="section-row undefined"
                        >
                          <span>
                            <span>
                              They can also be content
                            </span>
                             
                          </span>
                        </div>
                      </div>
                      <div
                        class="no-user-select"
                      >
                        <div
                          class="section-row undefined"
                        >
                          <span>
                            <span>
                              - Content text is grey
                            </span>
                             
                          </span>
                        </div>
                      </div>
                      <div
                        class="no-user-select"
                      >
                        <div
                          class="section-row undefined"
                        >
                          <span>
                            <span>
                              - Context text cannot have children
                            </span>
                             
                          </span>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="headline-dashplus"
                  >
                    <svg
                      height="16"
                      title="dash-collapse"
                      width="16"
                    >
                      <line
                        stroke="black"
                        stroke-width="1.5"
                        x1="8"
                        x2="16"
                        y1="8"
                        y2="8"
                      />
                    </svg>
                  </div>
                </div>
              </div>
              <div
                class="headline-row undefined"
                data-testid="headline"
                level="2"
              >
                <div
                  class="headline-row-item"
                >
                  <div>
                    <div
                      class="headline-star"
                    >
                      <svg
                        height="16px"
                        width="16px"
                      >
                        <circle
                          cx="50%"
                          cy="50%"
                          r="4px"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    class="headline-content"
                  >
                    <div
                      class="no-user-select"
                    >
                      <span>
                        <span>
                          You can sync your changes with an existing Dropbox account by clicking the options icon
                        </span>
                         
                      </span>
                    </div>
                    <div />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="headline-dashplus"
          >
            <svg
              height="16"
              title="dash-collapse"
              width="16"
            >
              <line
                stroke="black"
                stroke-width="1.5"
                x1="8"
                x2="16"
                y1="8"
                y2="8"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="headline-row undefined"
        data-testid="headline"
        level="1"
      >
        <div
          class="headline-row-item"
        >
          <div>
            <div
              class="headline-star"
            >
              <svg
                height="16px"
                width="16px"
              >
                <circle
                  cx="50%"
                  cy="50%"
                  r="4px"
                />
              </svg>
            </div>
          </div>
          <div
            class="headline-content"
          >
            <div
              class="no-user-select"
            >
              <span>
                <span>
                  Want to make a list of tasks?
                </span>
                 
              </span>
            </div>
            <div>
              <div
                class="headline-row undefined"
                data-testid="headline"
                level="2"
              >
                <div
                  class="headline-row-item"
                >
                  <div>
                    <div
                      class="headline-star"
                    >
                      <svg
                        height="16px"
                        width="16px"
                      >
                        <circle
                          cx="50%"
                          cy="50%"
                          r="4px"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    class="headline-content"
                  >
                    <div
                      class="no-user-select"
                    >
                      <span
                        class="headline-state-text red"
                      >
                         
                        TODO
                         
                      </span>
                      <span>
                        <span>
                          Mark a headline with TODO
                        </span>
                         
                      </span>
                    </div>
                    <div>
                      <div
                        class="no-user-select"
                      >
                        <div
                          class="section-row undefined"
                        >
                          <span>
                            <span>
                              Add some details about the task below in the content area
                            </span>
                             
                          </span>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="headline-dashplus"
                  >
                    <svg
                      height="16"
                      title="dash-collapse"
                      width="16"
                    >
                      <line
                        stroke="black"
                        stroke-width="1.5"
                        x1="8"
                        x2="16"
                        y1="8"
                        y2="8"
                      />
                    </svg>
                  </div>
                </div>
              </div>
              <div
                class="headline-row undefined"
                data-testid="headline"
                level="2"
              >
                <div
                  class="headline-row-item"
                >
                  <div>
                    <div
                      class="headline-star"
                    >
                      <svg
                        height="16px"
                        width="16px"
                      >
                        <circle
                          cx="50%"
                          cy="50%"
                          r="4px"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    class="headline-content"
                  >
                    <div
                      class="no-user-select"
                    >
                      <span
                        class="headline-state-text green"
                      >
                         
                        DONE
                         
                      </span>
                      <span>
                        <span>
                          When you're done you can mark the task done
                        </span>
                         
                      </span>
                    </div>
                    <div />
                  </div>
                </div>
              </div>
              <div
                class="headline-row undefined"
                data-testid="headline"
                level="2"
              >
                <div
                  class="headline-row-item"
                >
                  <div>
                    <div
                      class="headline-star"
                    >
                      <svg
                        height="16px"
                        width="16px"
                      >
                        <circle
                          cx="50%"
                          cy="50%"
                          r="4px"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    class="headline-content"
                  >
                    <div
                      class="no-user-select"
                    >
                      <span
                        class="headline-state-text red"
                      >
                         
                        TODO
                         
                      </span>
                      <span
                        class="headline-priority-text"
                      >
                         #[
                        A
                        ] 
                      </span>
                      <span>
                        <span>
                          Have a lot of todo tasks? Give them a priority
                        </span>
                         
                      </span>
                    </div>
                    <div />
                  </div>
                </div>
              </div>
              <div
                class="headline-row undefined"
                data-testid="headline"
                level="2"
              >
                <div
                  class="headline-row-item"
                >
                  <div>
                    <div
                      class="headline-star"
                    >
                      <svg
                        height="16px"
                        width="16px"
                      >
                        <circle
                          cx="50%"
                          cy="50%"
                          r="4px"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    class="headline-content"
                  >
                    <div
                      class="no-user-select"
                    >
                      <span
                        class="headline-state-text red"
                      >
                         
                        TODO
                         
                      </span>
                      <span
                        class="headline-priority-text"
                      >
                         #[
                        B
                        ] 
                      </span>
                      <span>
                        <span>
                          This task is less important than the one above
                        </span>
                         
                      </span>
                    </div>
                    <div />
                  </div>
                </div>
              </div>
              <div
                class="headline-row undefined"
                data-testid="headline"
                level="2"
              >
                <div
                  class="headline-row-item"
                >
                  <div>
                    <div
                      class="headline-star"
                    >
                      <svg
                        height="16px"
                        width="16px"
                      >
                        <circle
                          cx="50%"
                          cy="50%"
                          r="4px"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    class="headline-content"
                  >
                    <div
                      class="no-user-select"
                    >
                      <span
                        class="headline-state-text red"
                      >
                         
                        TODO
                         
                      </span>
                      <span
                        class="headline-priority-text"
                      >
                         #[
                        C
                        ] 
                      </span>
                      <span>
                        <span>
                          Priority integrates with Agenda mode, a feature that will be coming soon
                        </span>
                         
                      </span>
                    </div>
                    <div />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="headline-dashplus"
          >
            <svg
              height="16"
              title="dash-collapse"
              width="16"
            >
              <line
                stroke="black"
                stroke-width="1.5"
                x1="8"
                x2="16"
                y1="8"
                y2="8"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="headline-row undefined"
        data-testid="headline"
        level="1"
      >
        <div
          class="headline-row-item"
        >
          <div>
            <div
              class="headline-star"
            >
              <svg
                height="16px"
                width="16px"
              >
                <circle
                  cx="50%"
                  cy="50%"
                  r="4px"
                />
              </svg>
            </div>
          </div>
          <div
            class="headline-content"
          >
            <div
              class="no-user-select"
            >
              <span>
                <span>
                  Does Web-org support
                </span>
                 
              </span>
              <span>
                <span
                  class="text-bold"
                >
                  bold,
                </span>
                 
              </span>
              <span>
                <span
                  class="text-underline"
                >
                  underline,
                </span>
                 
              </span>
              <span>
                <span
                  class="text-italic"
                >
                  italic,
                </span>
                 
              </span>
              <span>
                <span>
                  and
                </span>
                 
              </span>
              <span>
                <span
                  class="text-strikethrough"
                >
                  strikethrough text?
                </span>
                 
              </span>
            </div>
            <div>
              <div
                class="no-user-select"
              >
                <div
                  class="section-row undefined"
                >
                  <span>
                    <span>
                      Yes, yes it does
                    </span>
                     
                  </span>
                </div>
              </div>
              <div
                class="no-user-select"
              >
                <div
                  class="section-row undefined"
                >
                  <span>
                    <span>
                      - wrap text with * for bold
                    </span>
                     
                  </span>
                </div>
              </div>
              <div
                class="no-user-select"
              >
                <div
                  class="section-row undefined"
                >
                  <span>
                    <span>
                      - wrap text with _ for underline
                    </span>
                     
                  </span>
                </div>
              </div>
              <div
                class="no-user-select"
              >
                <div
                  class="section-row undefined"
                >
                  <span>
                    <span>
                      - wrap text with / for italic
                    </span>
                     
                  </span>
                </div>
              </div>
              <div
                class="no-user-select"
              >
                <div
                  class="section-row undefined"
                >
                  <span>
                    <span>
                      - wrap text with + for strikethrough
                    </span>
                     
                  </span>
                </div>
              </div>
              <div
                class="no-user-select"
              >
                <div
                  class="section-row undefined"
                >
                  <span>
                    <span>
                        
                    </span>
                     
                  </span>
                </div>
              </div>
              <div
                class="headline-row undefined"
                data-testid="headline"
                level="2"
              >
                <div
                  class="headline-row-item"
                >
                  <div>
                    <div
                      class="headline-star"
                    >
                      <svg
                        height="16px"
                        width="16px"
                      >
                        <circle
                          cx="50%"
                          cy="50%"
                          r="4px"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    class="headline-content"
                  >
                    <div
                      class="no-user-select"
                    >
                      <span>
                        <span>
                          this is an add for test
                        </span>
                         
                      </span>
                    </div>
                    <div>
                      <div
                        class="no-user-select"
                      >
                        <div
                          class="section-row undefined"
                        >
                          <span>
                            <span>
                              here is some test content for the new note
                            </span>
                             
                          </span>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="headline-dashplus"
                  >
                    <svg
                      height="16"
                      title="dash-collapse"
                      width="16"
                    >
                      <line
                        stroke="black"
                        stroke-width="1.5"
                        x1="8"
                        x2="16"
                        y1="8"
                        y2="8"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="headline-dashplus"
          >
            <svg
              height="16"
              title="dash-collapse"
              width="16"
            >
              <line
                stroke="black"
                stroke-width="1.5"
                x1="8"
                x2="16"
                y1="8"
                y2="8"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
  <button
    aria-label="Add"
    class="mdc-fab"
    title="Add"
  >
    <i
      class="mdc-fab__icon material-icons mdc-fab-icon"
    >
      add
    </i>
  </button>
  <div
    class="mdc-bottom-app-bar__fab mdc-bottom-app-bar__fab--center-cut mdc-fab material-icons"
  />
  <div
    class="mdc-bottom-app-bar__row"
  >
    <section
      class="mdc-bottom-app-bar__section mdc-bottom-app-bar__section--align-start"
    >
      <button
        aria-label="Menu"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart"
        tabindex="0"
        title="toggle-file-explorer"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <h6
        class="MuiTypography-root MuiTypography-h6"
        data-testid="filename-titlebar"
      >
        Welcome to Weborg.org
      </h6>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        role="presentation"
        title="SettingsIcon"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
        />
      </svg>
    </section>
  </div>
</div>
`;
